content/docs/atom-guide/dev-guide/shaders/azsl/unbounded-arrays.md

```diff
@@ -0,0 +1,306 @@
+In this example, `m_texSRV_unbounded`, is the first texture SRV so it will start at **t0**, and take over the rest of the indices until **tN**. Any other texture SRV declared afterwards won't find a register index to bind to.  
+```cpp
+    ShaderResourceGroupSemantic slot1
+    {
+        FrequencyId = 1;
+    };
+    
+    struct MyStruct
+    {
+        float3 m_a;
+        float3 m_b;
+    };
+    
+    //For DX12, --unique-idx is OFF, so the expected register assignment is as follows
+    ShaderResourceGroup SRG1 : slot1
+    {
+        Texture2D<float4>        m_texSRV_unbounded[]; // Will be bound to t0 register, and will take over t1+.
+        Texture2D<float4>        m_texSRV1;      // ERROR. There's no tN available to bind this resource to.
+        Texture2D<float4>        m_texSRV2;      // ERROR. There's no tN available to bind this resource to.
+        RWTexture2D<float4>      m_texUAV1;      // Will be bound to u0 register.
+    	RWTexture2D<float4>      m_texUAV2;      // Will be bound to u1 register.
+        Sampler                  m_sampler1;     // Will be bound to s0 register.
+        Sampler                  m_sampler2;     // Will be bound to s1 register.
+        ConstantBuffer<MyStruct> m_cb1; // Will be bound to b1 register. (Because The ConstantBuffer reserved for SRG1 will be bound to b0 register).
+        ConstantBuffer<MyStruct> m_cb2; // Will be bound to b2 register.
+    };
+```
+### Example 3: (Error case)
+This error case is very similar to Example 2, but `m_texSRV_unbounded` is declared after `m_texSRV1`, but before `m_texSRV2`.  
+```cpp
+    ShaderResourceGroupSemantic slot1
+    {
+        FrequencyId = 1;
+    };
+    
+    struct MyStruct
+    {
+        float3 m_a;
+        float3 m_b;
+    };
+    
+    //For DX12, --unique-idx is OFF, so the expected register assignment is as follows
+    ShaderResourceGroup SRG1 : slot1
+    {
+        Texture2D<float4>        m_texSRV1;      // Will be bound to t0 register.
+        Texture2D<float4>        m_texSRV_unbounded[]; // Will be bound to t1 register, and will take over t2+.
+        Texture2D<float4>        m_texSRV2;      // ERROR. There's no tN available to bind this resource to.
+        RWTexture2D<float4>      m_texUAV1;      // Will be bound to u0 register.
+    	RWTexture2D<float4>      m_texUAV2;      // Will be bound to u1 register.
+        Sampler                  m_sampler1;     // Will be bound to s0 register.
+        Sampler                  m_sampler2;     // Will be bound to s1 register.
+        ConstantBuffer<MyStruct> m_cb1; // Will be bound to b1 register. (Because The ConstantBuffer reserved for SRG1 will be bound to b0 register).
+        ConstantBuffer<MyStruct> m_cb2; // Will be bound to b2 register.
+    };
+```
+### Example 4:
+This case fixes the issues in Example 3. By moving the point of declaration of `m_texSRV_unbounded[]` after `m_texSRV2` there are no more register assignment conflicts.  
```

Comment:
* Add a sentence to teach readers what the rule is that resolves the problem. In this case, it's moving `m_texSRV_unbounded[]` after `m_texSRV2`, but what about in my own shader code? What is the rule I apply? 
  Consider my suggestion for answering that: 

```suggestion
#### Solution

This example demonstrates how to resolve the register assignment conflicts in the previous error cases. To resolve this, define the unbounded array last among the group of resources with the same resource type. In this case, define `m_texSRV_unbounded` after `m_texSRV1` and `m_texSRV2`.
```